### ISSUE TITLE
Feature: Authorise with bearer token instead of basic auth

### ISSUE THREAD

**Issue by danielloader** [ezhov-evgeny/webdav-client-python-3#41]
Hey this is a great library and been using it extensively in scripts, but I would like to propose the option to authorise using a header, `Authorization: Bearer $token` instead of `Authorization: Basic $base64auth`.

Would this be feasible?

**Comment by ezhov-evgeny** [comment]
@danielloader I think it is possible I'll investigate it

**Comment by danielloader** [comment]
Lines 138 in client.py
Should really have checked!


        ```
    `webdav_login`: (optional) login name for WebDAV server can be empty in case using of token auth.
            `webdav_password`: (optional) password for WebDAV server can be empty in case using of token auth.
            `webdav_token': (optional) token for WebDAV server can be empty in case using of login/password auth.
```

**Comment by danielloader** [comment]
Put the request through mitmproxy, noticed the request header is malformed, doing `Authorization: Basic ` with some junk base64 encoding after.

**Comment by danielloader** [comment]
Created a proof of concept PR, works for me but would appreciate it being looked at as someone new to python.

https://github.com/ezhov-evgeny/webdav-client-python-3/pull/42

**Comment by ezhov-evgeny** [comment]
Merged into the develop branch. It will be published in version 3.14.1
